File: companies/spec/factories/company.rb
require 'faker'

FactoryBot.define do

 factory :company, class: Company do
   name {Faker::Company.name}
   city { Faker::Address.city}
   state {Faker::Address.state_abbr}
   description {Faker::Lorem.paragraphs(7).join}
   founded_date {Faker::Date.backward(14000)}

   factory :company_with_founders do
     transient do
       founders_count { 5 }
     end

     after(:create) do |company, evaluator|
       create_list(:founder, evaluator.founders_count, company: company)
     end
   end
 end 
end

File: fsc-backend/util/company.js
const formatCompanyPostData = data => ({
  name: data.name,
  city: data.city,
  state: data.state,
  description: data.description,
  foundedDate: data.foundedDate || null
});

const formatCompanyPatchData = (company, data) => {
  const newObject = {
    name: data.name ? data.name : company.name,
    city: data.city ? data.city : company.city,
    state: data.state ? data.state : company.state,
    description: data.description ? data.description : company.description,
    foundedDate: data.foundedDate ? data.foundedDate: company.foundedDate
  }
  return newObject;
}

module.exports = {
  formatCompanyPatchData,
  formatCompanyPostData
}
